cmd/ebrelayer/relayer/cosmos.go

```diff
@@ -94,114 +88,108 @@ func (sub CosmosSub) Start(txFactory tx.Factory, completionEvent *sync.WaitGroup
 
 	defer client.Stop() //nolint:errcheck
 
-	// Subscribe to all new blocks
-	query := "tm.event = 'NewBlock'"
-	results, err := client.Subscribe(context.Background(), "test", query, 1000)
-	if err != nil {
-		sub.SugaredLogger.Errorw("sifchain client failed to subscribe to query.",
-			errorMessageKey, err.Error(),
-			"query", query)
-		completionEvent.Add(1)
-		go sub.Start(txFactory, completionEvent, symbolTranslator)
-		return
-	}
-
-	defer func() {
-		if err := client.Unsubscribe(context.Background(), "test", query); err != nil {
-			sub.SugaredLogger.Errorw("sifchain client failed to unsubscribe query.",
-				errorMessageKey, err.Error())
-		}
-	}()
-
 	quit := make(chan os.Signal, 1)
 	signal.Notify(quit, syscall.SIGINT, syscall.SIGTERM)
 	defer close(quit)
 
-	var lastProcessedBlock int64
-
-	data, err := sub.DB.Get([]byte(cosmosLevelDBKey), nil)
-	if err != nil {
-		log.Println("Error getting the last cosmos block from level db", err)
-		lastProcessedBlock = 0
-	} else {
-		lastProcessedBlock = new(big.Int).SetBytes(data).Int64()
-	}
-
 	for {
 		select {
+		// Handle any errors
 		case <-quit:
 			log.Println("we receive the quit signal and exit")
 			return
+		default:
+			sub.CheckNonceAndProcess(txFactory, client)
+			time.Sleep(time.Second * cosmosSleepDuration)
+		}
+	}
+}
 
-		case e := <-results:
-			data, ok := e.Data.(tmTypes.EventDataNewBlock)
-			if !ok {
-				sub.SugaredLogger.Errorw("sifchain client failed to extract event data from new block.",
-					"EventDataNewBlock", fmt.Sprintf("%v", e.Data))
-			}
-			blockHeight := data.Block.Height
+// CheckNonceAndProcess check the lock burn nonce and process the event
+func (sub CosmosSub) CheckNonceAndProcess(txFactory tx.Factory,
+	client *tmclient.HTTP) {
 
-			// Just start from current block number if never process any block before
-			if lastProcessedBlock == 0 {
-				lastProcessedBlock = blockHeight
-			}
-			sub.SugaredLogger.Infow("new sifchain block witnessed")
+	// get lock burn nonce and start block number from cosmos
+	globalNonce, blockNumber, err := sub.GetGlobalNonceBlockNumberFromCosmos(sub.NetworkDescriptor, sub.ValidatorName)
+	if err != nil {
+		sub.SugaredLogger.Errorw("failed to get the lock burn nonce from cosmos rpc",
+			errorMessageKey, err.Error())
+		return
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
```

Comment:
Waiting for one second seems likely to fail in production, even if it works in tests